### PR TITLE
feat: add annotations value to datastore

### PR DESCRIPTION
**Pull request by killianmuldoon**
This allows adding annotations to the default datastore deployed through the helm chart. Specifically I'm looking to add the `helm.sh/resource-policy: keep` so helm doesn't delete this CR when we uninstall.

### CODE REVIEW

**Comment by tobiasgiese** [comment]
Maybe it is worth adding a `datastore.keep` value that adds the annotation. This makes it more explicit to the users that it is possible to keep the datastore.

**Review by prometherion — APPROVED** [review]
LGTM, once merged we can release the v0.9.0 with this feature.

**Pull request merged**
